x/feegrant/keeper/msg_server_test.go

```diff
@@ -42,6 +49,49 @@ func (suite *KeeperTestSuite) TestGrantAllowance() {
 			true,
 			"decoding bech32 failed",
 		},
+		{
+			"valid: grantee account doesn't exist",
+			func() *feegrant.MsgGrantAllowance {
+				grantee := "cosmos139f7kncmglres2nf3h4hc4tade85ekfr8sulz5"
+				granteeAccAddr := types.MustAccAddressFromBech32(grantee)
+				any, err := codectypes.NewAnyWithValue(&feegrant.BasicAllowance{
+					SpendLimit: suite.atom,
+					Expiration: &oneYear,
+				})
+
+				suite.accountKeeper.EXPECT().GetAccount(gomock.Any(), granteeAccAddr).Return(nil).AnyTimes()
+
+				acc := authtypes.NewBaseAccountWithAddress(granteeAccAddr)
+				suite.accountKeeper.EXPECT().NewAccountWithAddress(gomock.Any(), types.MustAccAddressFromBech32(grantee)).Return(acc).AnyTimes()
+				suite.accountKeeper.EXPECT().SetAccount(gomock.Any(), acc).Return()
+
+				suite.Require().NoError(err)
+				return &feegrant.MsgGrantAllowance{
+					Granter:   suite.addrs[0].String(),
+					Grantee:   grantee,
+					Allowance: any,
+				}
+			},
+			false,
+			"",
+		},
+		{
+			"invalid: past expiry",
+			func() *feegrant.MsgGrantAllowance {
+				any, err := codectypes.NewAnyWithValue(&feegrant.BasicAllowance{
+					SpendLimit: suite.atom,
+					Expiration: &yesterday,
+				})
+				suite.Require().NoError(err)
+				return &feegrant.MsgGrantAllowance{
+					Granter:   suite.addrs[0].String(),
+					Grantee:   suite.addrs[1].String(),
+					Allowance: any,
+				}
+			},
+			true,
+			"",
```

Comment:
nit: would be nice to add the error message here

Reply:
+1